### PR TITLE
react-syntax-highlighter: Update styles folder for v11.0.2

### PR DESCRIPTION
**Pull request by giuband**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/conorhastings/react-syntaax-highlighter/commit/115c16386384bc0dcdb49853b2865de7d074b361#diff-04c6e90faac2675aa89e2176d2eec7d8
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by guoyunhe — APPROVED** [review]
Good for me. Thanks for contributing and sorry for late review!

File: types/react-syntax-highlighter/index.d.ts [hunk 1]
```diff
@@ -106,824 +106,824 @@ declare module "react-syntax-highlighter/prism" {
     > {}
 }
 
-declare module "react-syntax-highlighter/dist/styles/hljs" {
+declare module "react-syntax-highlighter/dist/esm/styles/hljs" {
```
Comment (rbuckton) [inline types/react-syntax-highlighter/index.d.ts]: This seems like a breaking change for anyone still using v10.

**Review by rbuckton — COMMENTED** [review]
If there is a possibility that the v10 definitions may need to be changed in the future, I would recommend you create a `v10` folder with a copy of the 10.2 declarations per the instructions here: https://github.com/DefinitelyTyped/DefinitelyTyped#if-a-library-is-updated-to-a-new-major-version-with-breaking-changes-how-should-i-update-its-type-declaration-package.

If you do not believe that there is a need to maintain a separate v10 version for future updates, let me know and I will merge this PR.

**Comment by guoyunhe** [comment]
I think we probably won't have volunteers to maintain the types for v10. And the last v10 type is good enough for most usage. So I think we don't need to keep a v10 back up folder.

File: types/react-syntax-highlighter/index.d.ts (hunk 1 above)
Comment (stof) [reply types/react-syntax-highlighter/index.d.ts]: actually, this was a mistake in v10 types AFAICT, as the changelog of the lib mentions that as a breaking change between 9 and 10.

and this is missing some of the updates, as languages also moved.

**Pull request merged**